File: app/src/main/java/com/example/mvvmnavigation/MainActivity.java
package com.example.mvvmnavigation;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.example.mvvmnavigation.databinding.ActivityMainBinding;
import com.example.mvvmnavigation.databinding.PartResultBinding;

public class MainActivity extends AppCompatActivity {

  ActivityMainBinding mainBinding;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    mainBinding = ActivityMainBinding.inflate(getLayoutInflater());
    setContentView(mainBinding.getRoot());
  }

  public @Nullable ViewGroup getRoot(){
    return mainBinding.getRoot();
  }

  public PartResultBinding getPartResultLayout(){
    return PartResultBinding.bind(mainBinding.getRoot());
  }

}

File: app/src/main/java/com/example/mvvmnavigation/MainActivity.kt
package com.example.mvvmnavigation

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.mvvmnavigation.databinding.ActivityMainBinding
import com.example.mvvmnavigation.mvi_mvvm.MainScreenIntent
import com.example.mvvmnavigation.mvi_mvvm.MainViewModel

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private var model: MainViewModel? = null // todo: should be inited

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
//        val view = binding.root
        setContentView(binding.root)
        //model =        ViewModelProvider.Factory(this)
        binding.actionButton.setOnClickListener{
            model?.obtainIntent(MainScreenIntent.ClickButtonIntent)
        }
    }
}